server/index.js

```diff
@@ -68,6 +92,95 @@ app.get('/api/venues/:city/:search', (req, res, next) => {
       res.send(formatData(json));
     });
 });
+app.get('/create', (req, res, next) => {
+
+  connection.query("DELETE FROM favorites WHERE venue_json = '456'", function (err) {
+    if (err) {
+      console.error(err);
+    }
+    res.end('created');
+  });
+
+});
+app.get('/dbtest', (req, res, next) => {
+  connection.query('SELECT * FROM users', function (err, rows) {
+    if (err) {
+      console.error(err);
+    }
+    res.send(rows);
+  });
+});
+app.get('/api/user/favorites', (req, res, next) => {
+
+  connection.query('SELECT venue_json FROM favorites', (qErr, qRes) => {
+    if (qErr) {
+      console.error(qErr);
+    }
+    const data = [];
+    qRes.forEach(row => {
+      data.push(JSON.parse(row.venue_json));
+    });
+    res.send(data);
+
+  });
+
+});
+
+app.get('/api/get/allposts', (req, res, next) => {
+  connection.query('SELECT * FROM posts ORDER BY date_created DESC', (qErr, qRes) => {
+    if (qRes && qRes.rows) {
+      res.json(qRes.rows);
+    }
+
+    if (qErr) {
+      console.error(qErr);
+      res.json({ err: qErr });
```

Comment:
never send server errors to the client it's a security risk, only a 500 with a simple message